app/models/skin.rb

```diff
@@ -276,12 +279,14 @@ def get_all_parents
   
   # This is the main function that actually returns code to be embedded in a page
   def get_style(roles_to_include = DEFAULT_ROLES_TO_INCLUDE)
-    style = ""
-    if self.get_role != "override" && self.get_role != "site"
-      style += AdminSetting.default_skin != Skin.default ? AdminSetting.default_skin.get_style(roles_to_include) : (Skin.get_current_site_skin ? Skin.get_current_site_skin.get_style(roles_to_include) : '')
+    Rails.cache.fetch('skins_style_tags/v1/'+self.get_role+'/'+self.id.to_s+'/'+AdminSetting.default_skin.updated_at.to_s+'/'+Skin.default.updated_at.to_s+'/'+Skin.get_current_version+'/'+(Rails.cache.fetch('skins_generation') || '0')) do 
+      style = ""
+      if self.get_role != "override" && self.get_role != "site"
+        style += AdminSetting.default_skin != Skin.default ? AdminSetting.default_skin.get_style(roles_to_include) : (Skin.includes(:parent_skins).get_current_site_skin ? Skin.includes(:parent_skins).get_current_site_skin.get_style(roles_to_include) : '')
```

Comment:
I have not changed this line. I did spend the time to work out what it did however it has dropped out of my memory, I don't like the ? that much as it can make the code unreadable. I could refactor it if needed. However that adds risk.